core/src/main/java/hudson/model/Label.java

```diff
@@ -353,6 +353,48 @@ public boolean isOffline() {
         return true;
     }
 
+    /**
+     * Returns the icon for this label depends on state of all associated computers.
+     *
+     * see {@link #isOffline()}
+     * see {@link #getIconClassName()}
+     */
+    @Exported
+    public String getIcon() {
+        // All nodes are offline
+        if (isOffline()) return "symbol-computer-offline";
+        return "symbol-computer";
+    }
+
+    /**
+     * Returns the class name that will be used to lookup the icon.
+     *
+     * This class name will be added as a class tag to the html img tags where the icon should
+     * show up followed by a size specifier given by {@link Icon#toNormalizedIconSizeClass(String)}
+     * The conversion of class tag to src tag is registered through {@link IconSet#addIcon(Icon)}
```

Comment:
```suggestion
     * show up followed by a size specifier given by {@link org.jenkins.ui.icon.Icon#toNormalizedIconSizeClass(String)}
     * The conversion of class tag to src tag is registered through {@link org.jenkins.ui.icon.IconSet#addIcon(org.jenkins.ui.icon.Icon)}
```
Maybe this makes it more clear what I was referring to.